packages/api/src/modules/deposit-scdl-process/check/DepositScdlProcess.check.service.ts

```diff
@@ -0,0 +1,62 @@
+import { CreateDepositScdlLogDto, DepositScdlLogDto } from "dto";
+import { BadRequestError, ConflictError } from "core";
+import Siret from "../../../identifierObjects/Siret";
+
+export class DepositScdlProcessCheckService {
+    public validateCreate(dto: CreateDepositScdlLogDto) {
+        if (!dto.overwriteAlert) {
+            throw new BadRequestError("overwrite alert must be accepted");
+        }
+    }
+
+    public validateUpdateConsistency(depositScdlLogDto: DepositScdlLogDto, step: number) {
+        this.ensureExactPropertiesForStep(step, depositScdlLogDto);
+        this.validateFields(depositScdlLogDto);
+    }
+
+    private ensureExactPropertiesForStep(step: number, depositScdlLogDto: DepositScdlLogDto) {
+        const allowedPropsByStep: Record<number, (keyof DepositScdlLogDto)[]> = {
+            1: ["overwriteAlert"],
+            2: ["grantOrgSiret"],
+            3: ["permissionAlert"],
+        };
+
+        const allowedProps = allowedPropsByStep[step];
+        if (!allowedProps) {
+            throw new BadRequestError(`step ${step} is not supported`);
+        }
+
+        const dtoKeys = Object.entries(depositScdlLogDto)
+            .filter(([_, value]) => value !== undefined)
```

Comment:
```suggestion
            .filter(([_key, value]) => value !== undefined)
```

soyons descriptif, non ? c'est plus agréable à l'oeil (ça reste personnel)